lib/id3c/cli/command/etl/fhir.py

```diff
@@ -463,6 +476,22 @@ def process_patient_language(patient: Patient) -> Optional[str]:
     return matching_system_code(preferred_language[0].language, LANGUAGE_SYSTEM)
 
 
+def process_encounter_reason(encounter: Encounter) -> Optional[List[dict]]:
+    """
+    Returns the coding concept for all the reason codes of an *encounter*.
+    """
+    if not encounter.reasonCode:
+        return None
+
+    reason_codes: List[Coding] = []
+
+    for codeable_concept in encounter.reasonCode:
+        codes = codeable_concept.coding
+        reason_codes.extend(codes)
+
+    return [r.as_json() for r in reason_codes]
```

Comment:
FWIW, a more Pythonic approach here (which I do think is more clear) is a list comprehension:

```python
    return [
        coding.as_json()
        for concept in encounter.reasonCode
        for coding in concept.coding ]
```